lib/manager/pub/artifacts.ts

```diff
@@ -0,0 +1,91 @@
+import { TEMPORARY_ERROR } from '../../constants/error-messages';
+import { logger } from '../../logger';
+import { ExecOptions, exec } from '../../util/exec';
+import {
+  getSiblingFileName,
+  readLocalFile,
+  writeLocalFile,
+} from '../../util/fs';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+async function pubUpdate(
+  pubspecPath: string,
+  newPackageFileContent: string
+): Promise<void> {
+  let cmd = '';
+  if (newPackageFileContent.includes('sdk: flutter')) {
+    cmd = 'flutter pub upgrade';
+  } else {
+    cmd = 'pub upgrade';
+  }
```

Comment:
Does the flutter sdk includes the normal dart sdk?

Can we extract the used sdk version?  so we can provide a matching flutter sdk version?

Reply:
When flutter is setup, it contains the normal dart sdk. we can extract both the dart and flutter sdk versions:
```
$ flutter --version --machine
{
  "frameworkVersion": "2.2.3",
  "channel": "stable",
  "repositoryUrl": "https://github.com/flutter/flutter",
  "frameworkRevision": "f4abaa0735eba4dfd8f33f73363911d63931fe03",
  "frameworkCommitDate": "2021-07-01 12:46:11 -0700",
  "engineRevision": "241c87ad800beeab545ab867354d4683d5bfb6ce",
  "dartSdkVersion": "2.13.4",
  "flutterRoot": "/home/work/flutter"
}
```
For what would the information be used?